5/README.md

```diff
@@ -45,23 +53,72 @@ There are many use cases for the usage of a Property:
+* All device and class properties MAY be directly modified and accessed through the database device or its file
+* Attribute properties SHOULD be modified and accessed at any time through the device object (RFC-2)
+* All default attribute and device properties are OPTIONAL
+* The attribute, class and device properties MAY be defined in the device code and they MAY be OPTIONAL OR REQUIRED (mandatory)
+* TODO memorized attributes, system properties
+* TODO : properties history
+* TODO: error management
+* All device and class properties MUST be loaded at init command or device start-up
+* If a device property does not exist, the device MUST load the class property having the same name
+* All device and class properties MAY be modified and accessed through the database device when using a Tango Database (RFC-6)
+* A Tango device MUST manage the following default device properties: cmd_min_poll_period, min_poll_period, attr_min_poll_period, poll_ring_depth, cmd_poll_ring_depth, attr_poll_ring_depth, polled_attr, logging_target, logging_level, logging_rft (TODO: description of each one)
+* An attribute MUST manage the following default attribute properties (TODO: descriptions):
+	 * String **label** default value "";
+	 * String **description** default value  "No description"
+	 * String **unit** default value "No unit"
+	 * String **standard_unit** default value "No standard unit"
+	 * String **display_unit** default value "No display unit"
+	 * String **format** default value :
+				* attribute type is a string, "%s"
+				* attribute type is float or double, "%6.2f"
+				* "Not specified" otherwise
+  * String **min_value** default value "Not specified"
+  * String **max_value** default value  "Not specified"
+  * String **min_alarm** default value "Not specified"
+  * String **max_alarm** default value  "Not specified"
+  * String **min_warning** default value  "Not specified"
+  * String **max_warning** default value  "Not specified"
+  * String **delta_t** default value  "Not specified"
+  * String **delta_val** default value "Not specified"
+		* String **abs_change** default value "Not specified"
+		* String **rel_change** default value "Not specified"
+		* String **event_period** default value "Not specified"
+		* String **archive_period** default value "Not specified"
+		* String **archive_rel_change** default value "Not specified"
+		* String **archive_abs_change** default value "Not specified"
+		* String **enum_labels** default value "Not specified"
+		* String **__root_att** default value "Not specified"
+		* String **enum_labels**
+		* String **__value**
+
 
 ### Naming convention
 * The Property's Name SHALL use the following convention:
 ``` ABNF
 property-name = 1*VCHAR
 ```
+* The property name is case insensitive
+* The Property value MUST use the following convention:
+        single value that may contain any character
+        or an array with carriage return
+        special values: "NaN", "inf"
```

Comment:
So the property values are something like

```ABNF
propery-value = CHAR / 1*CHAR CR / "NaN" / "inf"
```

?

Reply:
I would say:

property-value = 1*[1*VCHAR | "nan" | "inf"] 

where VCHAR is the visible character set and nan and inf are case in-sensitive. The CR is part of the implementation in the database. I am not sure it belongs in the property value definition.

Reply:
`property-value = 1*[1*VCHAR | "nan" | "inf"]` does not respect the ABNF standard for me (https://tools.ietf.org/html/rfc5234 or https://en.wikipedia.org/wiki/Augmented_Backus%E2%80%93Naur_form). If the CR is part of the implementation in the database, how can we define arrays?